app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -110,7 +113,16 @@ def start_clone(clone_options)
       vim_clone_options[key] = ci.ems_ref_obj
     end
 
-    vim_clone_options[:datastore] = clone_options[:host].host_storages.find_by(:storage_id => clone_options[:datastore].id).ems_ref
+    host_ids = if clone_options[:host]
+                 clone_options[:host].id
+               else
+                 clone_options[:cluster].hosts.collect(&:id)
```

Comment:
Is `.hosts` an ActiveRecord Collection?  If so, `pluck(:id)` would be slightly more efficient.